addons/website_mass_mailing/static/src/js/website_mass_mailing.editor.js

```diff
@@ -98,11 +119,22 @@ options.registry.mailing_list_subscribe = options.Class.extend({
             }
         }
         const checkboxEl = document.createElement('we-checkbox');
-        checkboxEl.setAttribute('string', _t("Display Thanks Button"));
+        checkboxEl.setAttribute('string', this._getThanksButtonTitle());
         checkboxEl.dataset.toggleThanksButton = 'true';
         checkboxEl.dataset.noPreview = 'true';
         uiFragment.appendChild(checkboxEl);
     },
+    /**
+     * Gets the message to display in front of the "Toggle Thanks" button.
+     *
+     * @private
+     * @return {String} message to display.
+     */
+    _getThanksButtonTitle() {
+        return this.$target[0].dataset.newsletterTemplate === "form" ?
+            _t("Display Thanks Message") :
+            _t("Display Thanks Button");
+    },
```

Comment:
Even if we introduce the term in .po, all our non-English-speaking users will lose the translation of the option name (for months at least). Surely "Display Thanks Button" has already been translated into many languages. Even if it's not the ideal name when the template is a form, I'll leave it as it is in stable (but we can change in master). That way, all our users, whatever their language, will be able to understand what the option does.

So IMO, we shouldn't change that in stable and in master we can just rename it to "Display Thanks View" or something like that :+1:

Reply:
Same rationale as here: https://github.com/odoo/enterprise/pull/48973#pullrequestreview-1680057347 (don't mind the rest of the discussion, I don't remember it and I believe it was not translated anyway so it was off topic but still the point is the same)

> Sadly, it's not something we can do in this version as the term is translated here.
**A bad wording is always better than a wrong language.**
Imagine having an English word appearing all of a sudden on Chinese websites, whatever the translation and wording was, it will be better than a word in a language they don't even understand 🙂

There are people with different opinion so find a consensus between the two of you I'd say :)

Reply:
ChatGPT said:

> Considering the limitations, if the term "Display Thanks Button" is causing significant confusion or misunderstanding, then fixing it for clarity in English might be the way to go. However, be aware of the potential confusion it might introduce for non-English speakers. It's a tricky situation without a perfect solution.

I agree with it: it's a case by case choice but in this case, loosing the translation is probably not worth it :wink: It's not that confusing here, and hopefully the translation is not more confusing either. It's actually more confusing to have a "Thanks button" in the first place :joy: